docs/zephyr/quickref.rst

```diff
@@ -0,0 +1,151 @@
+    pin.off()                           # set pin to low
+
+    pin = Pin(("GPIO_1", 21), Pin.IN)   # create input pin on GPIO1
+
+    pin = Pin(("GPIO_1", 21), Pin.OUT, value=1)         # set pin high on creation
+
+    pin = Pin(("GPIO_1", 21), Pin.IN, Pin.PULL_UP)      # enable internal pull-up resistor
+
+
+    switch = Pin(("GPIO_2", 6), Pin.IN)
+    SW3 = Pin(("GPIO_0", 4), Pin.IN)
+
+    switch.irq(lambda t: print("SW2 changed"))
+    SW3.irq(lambda t: print("SW3 changed"))
+
+    ...
+
+Hardware I2C bus
+----------------
+
+Hardware I2C is accessed via the :ref:`machine.I2C <machine.I2C>` class::
+
+    from machine import I2C
+
+    i2c = I2C("I2C_0")          # construct an i2c bus
+    print(i2c)                  # print device name
+
+    i2c.scan()                  # scan the device for available I2C slaves
+
+    i2c.readfrom(0x1D, 4)                # read 4 bytes from slave 0x1D
+    i2c.readfrom_mem(0x1D, 0x0D, 1)      # read 1 byte from slave 0x1D at slave memory 0x0D
+
+    i2c.writeto(0x1D, b'abcd')           # write to slave with address 0x1D
+    i2c.writeto_mem(0x1D, 0x0D, b'ab')   # write to slave 0x1D at slave memory 0x0D
+
+    buf = bytearray(8)                  # create buffer of size 8
+    i2c.writeto(0x1D, b'abcd')          # write buf to slave 0x1D
+
+Hardware SPI bus
+----------------
+
+Hardware SPI is accessed via the :ref:`machine.SPI <machine.SPI>` class::
+
+    from machine import SPI
+
+    spi = SPI("SPI_0")          # construct a spi bus with default configuration
+    spi.init(baudrate=100000, polarity=0, phase=0, bits=8, firstbit=SPI.MSB) # set configuration
+
+    # equivalently, construct spi bus and set configuration at the same time
+    spi = SPI("SPI_0", baudrate=100000, polarity=0, phase=0, bits=8, firstbit=SPI.MSB)
+    print(spi)                  # print device name and bus configuration
+
+    spi.read(4)                 # read 4 bytes on MISO
+    spi.read(4, write=0xF)      # read 4 bytes while writing 0xF on MOSI
+
+    buf = bytearray(8)          # create a buffer of size 8
+    spi.readinto(buf)           # read into the buffer (reads number of bytes equal to the buffer size)
+    spi.readinto(buf, 0xF)      # read into the buffer while writing 0xF on MOSI
+
+    spi.write(b'abcd')          # write 4 bytes on MOSI
+
+    buf = bytearray(4)                  # create buffer of size 8
+    spi.write_readinto(b'abcd', buf)    # write to MOSI and read from MISO into the buffer
+    spi.write_readinto(buf, buf)        # write buf to MOSI and read back into the buf
+
+USocket
+-------
+
+module for networking (IPv4/IPv6)::
+
+    ...
+
+Disk Access
+-----------
+
+Use the :ref:`zephyr.DiskAccess <zephyr.DiskAccess>` class::
+
+    from zephyr import DiskAccess
+
+    disk = DiskAccess(*disk_name*)      # create disk object
+    print(disk)                         # print disk name
+
+    disk.readblocks(*block num*, *buffer*, *offset*)
```

Comment:
would be better to give examples of how to format and mount a filesystem, using `uos.VfsLfs2.mkfs` etc